tools/aliveness/src/main.rs

```diff
@@ -0,0 +1,358 @@
+use std::{
+    collections::HashMap,
+    env::args,
+    mem::size_of,
+    net::{Ipv4Addr, SocketAddr, SocketAddrV4},
+    ptr::read,
+    time::Duration,
+};
+
+use aliveness::{
+    service_manager::SystemServices, AlivenessState, BEACON_HEADER, BEACON_MULTICAST_GROUP,
+    BEACON_PORT,
+};
+use color_eyre::eyre::{bail, eyre, ContextCompat, Result, WrapErr};
+use configparser::ini::Ini;
+use futures_util::stream::StreamExt;
+use hula_types::{Battery, RobotState};
+use log::{error, info};
+use tokio::{
+    io::AsyncReadExt,
+    net::{UdpSocket, UnixStream},
+    select, spawn,
+    sync::{mpsc, oneshot},
+    task::JoinHandle,
+    time::interval,
+};
+use tokio_util::sync::CancellationToken;
+use zbus::{
+    zvariant::{OwnedObjectPath, Value},
+    Connection, MatchRule, MessageStream, MessageType, Proxy,
+};
+
+const HULA_SOCKET_PATH: &str = "/tmp/hula";
+const HULA_RETRY_TIMEOUT: Duration = Duration::from_secs(5);
+const OS_RELEASE_PATH: &str = "/etc/os-release";
+
+struct HulaService {
+    sender: mpsc::Sender<oneshot::Sender<RobotState>>,
+}
+
+impl HulaService {
+    fn create_sender() -> mpsc::Sender<oneshot::Sender<RobotState>> {
```

Comment:
This introduces some code duplication as we also have a HULA connector in the `HulaInterface` in the `hulk` crate. Maybe rename the `hula_types` crate to `hula` and also move the socket stuff and binary buffer casting into there to increase code reuse?

Reply:
Since the we only use hula to retrieve the body/head IDs and the battery status, this is planned to be replaced by a much more performant RPC interface to hula. That's why I would like to leave this for now as is and rather focus on the RPC interface in the future.

Reply:
Ok, fine for me.